blockstore/idstore.go

```diff
@@ -102,6 +102,14 @@ func (b *idstore) Put(ctx context.Context, blk blocks.Block) error {
 	return b.bs.Put(ctx, blk)
 }
 
+func (b *idstore) ForEachKey(f func(cid.Cid) error) error {
```

Comment:
Needed so that badge backed cold store satisfies the interface needed for pruning